python/mscclpp/language/rank.py

```diff
@@ -1,33 +1,60 @@
-# Copyright (c) Microsoft Corporation.
-# Licensed under the MIT License.
-
-from dataclasses import dataclass, field
-from typing import Dict
-
-
-class BarrierInfo:
-    def __init__(self, tb_list):
-        self.tb_list = tb_list
-
-    def __eq__(self, other):
-        return self.tb_list == other.tb_list
-
-    def __hash__(self):
-        return hash(tuple(self.tb_list))
+from mscclpp.language.internal.types import BufferType, Chunk
+from mscclpp.language.internal.operations import CopyOperation, LocalChunk
+from mscclpp.language.internal.globals import get_program
+from dataclasses import dataclass
 
 
 @dataclass
 class Rank:
-    rank_id: int
-    current_max_barrier_id: int = 0
-    current_barriers: Dict[BarrierInfo, int] = field(default_factory=dict)
-
-    def get_barrier_id(self, tb_list):
-        barrier_info = BarrierInfo(tb_list)
-        if barrier_info in self.current_barriers:
-            return self.current_barriers[barrier_info]
-        else:
-            self.current_barriers[barrier_info] = self.current_max_barrier_id
-            barrier_id = self.current_max_barrier_id
-            self.current_max_barrier_id += 1
-            return barrier_id
+    rank: int
+
+    def __init__(self, rank: int):
+        self.rank = rank
+        if rank >= get_program().num_ranks:
+            raise RuntimeError(f"Rank {rank} is out of bounds. Number of ranks: {self.prog.num_ranks}")
+
+    def get_input_buffer(self):
+        buffer_size = get_program().buffers_size[self.rank][BufferType.input]
+        return BaseBuffer(self.rank, BufferType.input, 0, buffer_size)
+
+    def get_output_buffer(self):
```

Comment:
Seems it will create new buffer object when call `get_output_buffer` function. Will it cause some problem?